src/protocols/rdp/channels/rdpdr/rdpdr.c

```diff
@@ -38,6 +38,17 @@ void guac_rdpdr_process_receive(guac_rdp_common_svc* svc,
     int component;
     int packet_id;
 
+    /* 
+     * Check that device redirection stream contains at least 4 bytes
+     * (UINT16 + UINT16).
+     */
+    if (Stream_GetRemainingLength(input_stream) < 4) {
+        guac_client_log(svc->client, GUAC_LOG_WARNING, "Device redirection "
+                "channel receive Stream does not contain the expected number "
+                "of bytes. Device redirection may not function as expected.");
```

Comment:
> Device redirection channel receive Stream ...

As phrased, this does not make sense to me. Looking at the context, this is guarding the read operation around the header common to all RDPDR PDUs, so I would suggest rephrasing this in that respect ("Device redirection channel PDU header" or "RDPDR channel PDU header" or similar).